app/models/messaging/bsnl/dlt_template.rb

```diff
@@ -23,11 +26,31 @@ def initialize(dlt_template_name)
     @status = STATUSES[details["Template_Status"]]
     @is_unicode = details["Is_Unicode"]
     @keys = details["Template_Keys"]
+    @version = details["Version"]
+    @is_latest_version = details["Is_Latest_Version"]
     @max_length_permitted = details["Max_Length_Permitted"].to_i
     @non_variable_text_length = details["Non_Variable_Text_Length"].to_i
     @variable_length_permitted = max_length_permitted - non_variable_text_length
   end
 
+  def self.latest_name_of(dlt_template_name)
+    BSNL_TEMPLATES.dig(drop_version_number(dlt_template_name), "Latest_Template_Version")
+  end
+
+  def self.drop_version_number(dlt_template_name)
+    dlt_template_name.chomp(".#{version_number(dlt_template_name)}")
+  end
+
+  def self.version_number(dlt_template_name)
+    version = dlt_template_name.split(".").last
+
+    if numeric?(version)
+      version.to_i
+    else
+      DEFAULT_VERSION_NUMBER
```

Comment:
If we return `1` here, won't the `latest_name_of` method break for all templates that have only a single version and don't have a numeric suffix in the template name?

Reply:
The `latest_name_of` looks at templates after removing the version from the template name, so it will find out the first version without the name as the "default".